app/controllers/lead_providers/base_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class BaseController < ApplicationController
+    include Pundit
+
+    before_action :authenticate_user!
+    before_action :ensure_lead_provider_or_admin
+
+  private
+
+    def ensure_lead_provider_or_admin
+      raise Pundit::NotAuthorizedError, "Forbidden" unless current_user&.admin? || current_user&.lead_provider?
```

Comment:
Unless can be confusing with compound predicates.

Reply:
I'm usually not in favour using unless - though in this example I think it works because of the name of the method and that is a one liner. We could use guard statement for example return if admin or lead provider - would that work with you?

Reply:
Yes, guards would be fine with me

Reply:
Done